docs/developers/how-to-test-if.md

```diff
@@ -1 +1,151 @@
+```
+
+
+## Setting up your test file
+
+You will need to import your plugin so that it can be instantiated and tested. You will also need some elements from `jest/globals`:
+For example, these are the improts for our `teads-curve` plugin.
+
+```ts
+import {describe, expect, jest, test} from '@jest/globals';
+import {TeadsCurveModel} from '../../../../lib';
+```
+
+You may require other imports for your specific set of tests.
+
+## Describe
+
+Each method should have its own dedicated `describe` block. 
+
+For model plugins there are a minimum set of methods defined by the `PluginInterface`: `configure()` and `execute()`. therefore, your unit tests should have *at least* two `describe` blocks for each of those methods:
+
+```ts
+describe("configure", ()=> {})
+describe("execute", ()=> {})
+```
+
+There should also be a `describe` blocks for successful plugin initialization.
+
+For example, here is a describe block checking that the `co2js` model initializes correctly:
+
+```typescript
+describe('init Co2jsModel: ', () => {
+  it('initalizes object with properties.', async () => {
+    expect(outputModel).toHaveProperty('configure');
+    expect(outputModel).toHaveProperty('execute');
+  });
+});
+```
+
+## It
+
+Within each `describe` block, each effect to be tested should have a dedicated `it` block.
+
+Here's an example of a new `describe` block for the `execute()` method on the `sci` model. The `describe` block indicates that we are testing effects of the `execute()` method. `it` is specific to a single outcome - in this case `it` tests that the plugin throws an exception if the user has provided invalid config data to the `time-sync` plugin, specifically that the user-provided `end-date` is before the user-provided `start-date`:
+
+```typescript
+describe('execute():', () => {
+
+  it('throws error if end is before start in global config.', async () => {
+    const basicConfig = {
+      'start-time': '2023-12-12T00:00:10.000Z',
+      'end-time': '2023-12-12T00:00:00.000Z',
+      interval: 5,
+      'allow-padding': true,
+    };
+
+    const timeModel = await new TimeSyncModel().configure(basicConfig);
+
+    try {
+      await timeModel.execute([
+        {
+          timestamp: '2023-12-12T00:00:00.000Z',
+          duration: 15,
+          'cpu-util': 10,
+        },
+        {
+          timestamp: '2023-12-12T00:00:10.000Z',
+          duration: 30,
+          'cpu-util': 20,
+        },
+      ]);
+    } catch (error) {
+      expect(error).toStrictEqual(
+        new InputValidationError('Start time or end time is missing.')
+      );
+    }
+  });
+})
+```
+
+
+## Errors
+
+Expect errors rather than catching them!
+
+```ts
+it("When no product name, it throws error 400", async () => {
+  let errorWeExceptFor = null;
+  try {
+    const result = await addNewProduct({});
+  } catch (error) {
+    expect(error.code).to.equal("InvalidInput");
+    errorWeExceptFor = error;
+  }
+  expect(errorWeExceptFor).not.to.be.null;
```

Comment:
let's write expect in catch block, 
let errorWeExceptFor = null; it's a side effect